frontend/src/components/dashboard/dashboard.tsx

```diff
@@ -3,59 +3,60 @@ import Search from "@/components/search/search";
 import AdminTable from "@/components/adminTable/table";
 import ButtonsRow from "@/components/buttonsRow/buttonsRow";
 import User from "@/types/user";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import axios from "@/axios/client";
 
-const users: User[] = [
-  {
-    id: 1,
-    name: "Leanne Graham",
-    username: "Bret",
-    email: "leannegraham@lynxlocks.com",
-    timeIn: "00:00:00",
-    date: "2024/01/01",
-  },
-  {
-    id: 2,
-    name: "Ervin Howell",
-    username: "Antonette",
-    email: "ervinhowell@lynxlocks.com",
-    timeIn: "00:00:00",
-    date: "2024/01/01",
-  },
-  {
-    id: 3,
-    name: "Clementine Bauch",
-    username: "Samantha",
-    email: "clementinebauch@lynxlocks.com",
-    timeIn: "00:00:00",
-    date: "2024/01/01",
-  },
-  {
-    id: 4,
-    name: "Patricia Lebsack",
-    username: "Karianne",
-    email: "patricialebsack@lynxlocks.com",
-    timeIn: "00:00:00",
-    date: "2024/01/01",
-  },
-];
-
 export default function Dashboard() {
+  const [users, setUsers] = useState<User[]>([]);
+  const [searchInput, setSearchInput] = useState("");
+
+  const updateUser = async (user: User) => {
+    // TODO: axios to update user
+    const newUsers = users.map((u) => {
+      if (u.id === user.id) {
+        return user;
+      }
+      return u;
+    });
+    setUsers(newUsers);
+  };
+
   useEffect(() => {
-    const f = async () => {
-      const users = axios.get("/users");
+    const getUsers = async () => {
+      const users = await axios.get("/users");
+      setUsers(
+        users.data.map((user: User) => {
+          user.timeIn = new Date().toLocaleTimeString(); // TODO: parse these from response
+          user.lastDateIn = new Date().toLocaleDateString();
```

Comment:
Docker containers use UTC by default, so this will not be real local time.
I think it's fine to leave for this PR, but can you please put a TODO here?
```diff
        users.data.map((user: User) => {
+          // TODO obtain timezone from TZ environment variable
          user.timeIn = new Date().toLocaleTimeString(); // TODO: parse these from response
          user.lastDateIn = new Date().toLocaleDateString();
```

Reply:
Sure, ill just add it onto the TODO that's already there. Basically these values should be stored on the user object. I guess if we store it as a unix timestamp (which we should), we still need to do the timezone logic.